## BioInformatics and Machine Learning Intern
Installing pyreadr to read rds file

In [1]:
!pip install pyreadr


     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 434.8/434.8 kB 5.9 MB/s eta 0:00:00


Upload the file and view the first five rows of the dataset

In [2]:
import pyreadr
from google.colab import files
import pandas as pd

# Prompt the user to upload an RDS file
uploaded = files.upload()

# Get the first uploaded file (assuming it's an RDS file)
rds_file_path = list(uploaded.keys())[0]

# Use the pyreadr.read_r() function to read the RDS file
result = pyreadr.read_r(rds_file_path)
keys_list = list(result.keys())
data = result[keys_list[0]]  # Use the first key (change as needed)

# Convert the data to a DataFrame
df = pd.DataFrame(data)

df.head()

Saving exgr_test.rds to exgr_test.rds


transcript_id    transcript_name sequence_names  start    end  width  \
0              1  ENST00000456328.2           chr1  11869  12227    359   
1              1  ENST00000456328.2           chr1  12613  12721    109   
2              1  ENST00000456328.2           chr1  13221  14409   1189   
3              2  ENST00000450305.2           chr1  12010  12057     48   
4              2  ENST00000450305.2           chr1  12179  12227     49   

  strand  exon_id          exon_name  rank  
0      +        1  ENSE00002234944.1     1  
1      +        5  ENSE00003582793.1     2  
2      +        8  ENSE00002312635.1     3  
3      +        2  ENSE00001948541.1     1  
4      +        3  ENSE00001671638.2     2

## Task 1: Counting Unique Transcripts
To find the count of unique transcripts, we identify the unique transcript IDs within the dataset. Each transcript is uniquely represented by its transcript ID

In [3]:
unique_transcripts = df['transcript_id'].nunique()
unique_transcripts


234485

As you can see there are 234485 transcripts in the entire datatset

##Task 2: Counting Unique Exons
### To determine the number of unique exons, we follow these steps:



1.  Group the data by transcript IDs.
2.  Calculate the count of exons within each transcript.
3. Sum up the counts from all transcripts to obtain the total count of unique exons.



In [4]:
# Group the DataFrame by transcript IDs
transcript_groups = df.groupby('transcript_id')

# Calculate the count of exons within each transcript
exon_counts_per_transcript = transcript_groups.size()

# Sum up the counts to find the total count of unique exons
total_unique_exons = exon_counts_per_transcript.sum()

# Display the total count of unique exons
total_unique_exons


1460986

There are 140986 total unique exons in the entire dataset.

### Task 3: Calculating Average and Median Exon Length
To calculate the average and median length of an exon, we utilize the 'mean' and 'median' functions applied to the 'width' column, which represents the exon width.



In [6]:
average_length = df['width'].mean()
print(average_length)

median_length = df['width'].median()
print(median_length)

262.9930635885628
130.0


The aveage length and median of an exon are 262.99 and 130 units respectively